Add test instructions to contributor guide

It would be useful to add instructions to run the tests to the contributors guide.
Especially because I also failed to run the tests :stuck_out_tongue: .

I tried the following:
```
python3.6 -m venv ./venv
source venv/bin/activate
pip install -r requirements.txt
pip install -r test-requirements.txt
python setup.py install
pytest ./tests
```

Which results in a lot of import errors similar to this one:
```
_________________________________________________________________ ERROR collecting tests/test_subgraphs.py __________________________________________________________________
ImportError while importing test module '/home/anton/gitrepos/flowpipe/tests/test_subgraphs.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/test_subgraphs.py:3: in <module>
    from flowpipe import Graph
E   ImportError: cannot import name 'Graph'
```

--- a/contributing.md
+++ b/contributing.md
@@ -28,6 +28,16 @@ We have an extensive, reliable test suite and we want to keep it that way, so pl
 We also want to keep the coverage at 100%. If there are good reasons for not covering parts of the code, please explicitely exclude them either via `# pragma: no cover` directly in the code or by specifying it in the [.coveragerc](.coveragerc) file.
 The tests have to pass on travis (py2.7 and py3.6).
 
+Execute the following steps to run the tests locally:
+```
+python3.6 -m venv ./venv
+source venv/bin/activate
+pip install -r requirements.txt
+pip install -r test-requirements.txt
+python setup.py develop
+pytest ./tests
+```
+
 ## Coding styleguide
 
 We generally follow [pep8](https://www.python.org/dev/peps/pep-0008/) with these additional requirements:

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -5,6 +5,16 @@ from flowpipe.plug import InputPlug, OutputPlug
 from flowpipe.graph import Graph
 
 class NodeForTesting(INode):
+    """
+    +---------------------+
+    |   NodeForTesting    |
+    |---------------------|
+    o in1<>               |
+    o in2<>               |
+    |                 out o
+    |                out2 o
+    +---------------------+
+    """
 
     def __init__(self, name=None, in1=None, in2=None, **kwargs):
         super(NodeForTesting, self).__init__(name, **kwargs)
@@ -19,6 +29,24 @@ class NodeForTesting(INode):
 
 @pytest.fixture
 def branching_graph():
+    """
+    +------------+          +------------+          +--------------------+
+    |   Start    |          |   Node2    |          |        End         |
+    |------------|          |------------|          |--------------------|
+    o in1<0>     |     +--->o in1<>      |          % in1                |
+    o in2<0>     |     |    o in2<0>     |     +--->o  in1.1<>           |
+    |        out o-----+    |        out o-----|--->o  in1.2<>           |
+    |       out2 o     |    |       out2 o     |    o in2<0>             |
+    +------------+     |    +------------+     |    |                out o
+                       |    +------------+     |    |               out2 o
+                       |    |   Node1    |     |    +--------------------+
+                       |    |------------|     |
+                       +--->o in1<>      |     |
+                            o in2<0>     |     |
+                            |        out o-----+
+                            |       out2 o
+                            +------------+
+    """
     graph = Graph(name="TestGraph")
     start = NodeForTesting(name='Start', graph=graph)
     n1 = NodeForTesting(name='Node1', graph=graph)

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,0 +1,32 @@
+import pytest
+
+from flowpipe.node import INode, Node
+from flowpipe.plug import InputPlug, OutputPlug
+from flowpipe.graph import Graph
+
+class NodeForTesting(INode):
+
+    def __init__(self, name=None, in1=None, in2=None, **kwargs):
+        super(NodeForTesting, self).__init__(name, **kwargs)
+        OutputPlug('out', self)
+        OutputPlug('out2', self)
+        InputPlug('in1', self, in1)
+        InputPlug('in2', self, in2)
+
+    def compute(self, in1, in2):
+        """Multiply the two inputs."""
+        return {'out': in1 * in2, 'out2': None}
+
+@pytest.fixture
+def branching_graph():
+    graph = Graph(name="TestGraph")
+    start = NodeForTesting(name='Start', graph=graph)
+    n1 = NodeForTesting(name='Node1', graph=graph)
+    n2 = NodeForTesting(name='Node2', graph=graph)
+    end = NodeForTesting(name='End', graph=graph)
+    start.outputs['out'] >> n1.inputs['in1']
+    start.outputs['out'] >> n2.inputs['in1']
+    n1.outputs['out'] >> end.inputs['in1']['1']
+    n2.outputs['out'] >> end.inputs['in1']['2']
+
+    yield graph

--- a/tests/test_graph.py
+++ b/tests/test_graph.py
@@ -157,25 +157,6 @@ def test_complex_branching_evaluation_sequence(clear_default_graph):
 
 
 def test_serialize_graph_to_json(clear_default_graph, branching_graph):
-    """
-    +------------+          +------------+          +--------------------+
-    |   Start    |          |   Node2    |          |        End         |
-    |------------|          |------------|          |--------------------|
-    o in1<0>     |     +--->o in1<>      |          % in1                |
-    o in2<0>     |     |    o in2<0>     |     +--->o  in1.1<>           |
-    |        out o-----+    |        out o-----|--->o  in1.2<>           |
-    |       out2 o     |    |       out2 o     |    o in2<0>             |
-    +------------+     |    +------------+     |    |                out o
-                       |    +------------+     |    |               out2 o
-                       |    |   Node1    |     |    +--------------------+
-                       |    |------------|     |
-                       +--->o in1<>      |     |
-                            o in2<0>     |     |
-                            |        out o-----+
-                            |       out2 o
-                            +------------+
-    """
-
     serialized = branching_graph.to_json()
     deserialized = Graph.from_json(serialized)
 

--- a/tests/test_graph.py
+++ b/tests/test_graph.py
@@ -156,7 +156,7 @@ def test_complex_branching_evaluation_sequence(clear_default_graph):
     assert 'end' == seq[-1]
 
 
-def test_serialize_graph_to_json(clear_default_graph):
+def test_serialize_graph_to_json(clear_default_graph, branching_graph):
     """
     +------------+          +------------+          +--------------------+
     |   Start    |          |   Node2    |          |        End         |
@@ -175,20 +175,11 @@ def test_serialize_graph_to_json(clear_default_graph):
                             |       out2 o
                             +------------+
     """
-    graph = Graph(name="TestGraph")
-    start = NodeForTesting(name='Start', graph=graph)
-    n1 = NodeForTesting(name='Node1', graph=graph)
-    n2 = NodeForTesting(name='Node2', graph=graph)
-    end = NodeForTesting(name='End', graph=graph)
-    start.outputs['out'] >> n1.inputs['in1']
-    start.outputs['out'] >> n2.inputs['in1']
-    n1.outputs['out'] >> end.inputs['in1']['1']
-    n2.outputs['out'] >> end.inputs['in1']['2']
 
-    serialized = graph.to_json()
+    serialized = branching_graph.to_json()
     deserialized = Graph.from_json(serialized)
 
-    assert graph.name == deserialized.name
+    assert branching_graph.name == deserialized.name
     assert serialized == deserialized.to_json()
 
 

